slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbSlashingProtection.java

```diff
@@ -247,6 +251,9 @@ public boolean maySignBlock(
 
   @Override
   public void prune() {
+    if (dbPruner.isEmpty()) {
+      throw new IllegalStateException("prune() called when dbPruner isn't initialized");
+    }
```

Comment:
Feels a bit messy having a prune method check whether prune is enabled, would be better IMO to Decorate DbSlashingProtection with the pruning functionality, e.g. wrap it in PruningDbSlashingProtection and add the prune method to that. That would get rid of all the optionals then too.

Reply:
Spiked on this please see https://github.com/ConsenSys/web3signer/pull/820

Reply:
Closing this PR in favour of #823